test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        
+        if(inputCommand.equals("add") || inputCommand.equals("clear") || inputCommand.equals("delete")) {
```

Comment:
You have found the right place to update the test. Now think about how to elegantly do it. Since you are checking the command types, consider what class helps you do this in the original code, which returns you a Command object. After that, you can test the isMutating() method directly by applying it to the Command object. Lastly, think about how to test the else branch, i.e. when the method is not mutating.

Reply:
I think that the else branch is not really needed. 
```
        //Confirm the state of data is as expected
        assertEquals(expectedAddressBook, addressBook);
        assertEquals(lastShownList, logic.getLastShownList());
```
already does the work.

Reply:
@arturs68 I understand your point. The rationale of doing the check for the else branch is actually a bit demanding in this case. It actually checks whether the file is saved again after a non-mutating command. Checking the content(even it matches the original content) does not  confirm that the saving does not happen. A file can be repeatedly saved even when no change is made. The point to implement the isMutating() method is also to achieve more efficiency so that saving only happens when the command is mutating. To test the else branch, you should check the timestamp of the storage file instead of its content, i.e. checking the last modified time for instance. Hope this clarifies my point:)